cartoframes/data/utils.py

```diff
@@ -0,0 +1,150 @@
+import time
+import binascii as ba
+
+from warnings import warn
+from carto.exceptions import CartoException, CartoRateLimitException
+
+from ..columns import Column
+
+try:
+    import geopandas
+    HAS_GEOPANDAS = True
+except ImportError:
+    HAS_GEOPANDAS = False
+
+
+DEFAULT_RETRY_TIMES = 3
+
+GEOM_COLUMN_NAMES = [
+    'geometry',
+    'the_geom',
+    'wkt_geometry',
+    'wkb_geometry',
+    'geom',
+    'wkt',
+    'wkb'
+]
+
+LAT_COLUMN_NAMES = [
+    'latitude',
+    'lat'
+]
+
+LNG_COLUMN_NAMES = [
+    'longitude',
+    'lng',
+    'lon',
+    'long'
+]
+
+
+def compute_query(dataset):
+    if dataset._table_name and dataset._schema:
+        return 'SELECT * FROM "{0}"."{1}"'.format(dataset._schema, dataset._table_name)
+
+
+def compute_geodataframe(dataset):
+    if dataset._df is not None:
+        df = dataset._df.copy()
```

Comment:
```suggestion
        df = dataset.dataframe.copy()
```

Reply:
I don't like to copy it. It could be a performance problem (in memory). I am thinking about it.

Reply:
If we don't copy the df to generate the gdf, the original df will be modified with an extra column. I'm not sure if we want this or not. Maybe it's a good thing to have the geometry column already there.

Reply:
Yeah, I worry about performance too. E.g., if a user has a dataframe that's 1GB, which isn't uncommon if they have complex polygons.

Reply:
What about a warning (or logging.info) that a new column is being added to the original dataframe?

Reply:
We have removed the copy() by now. But I think we should revisit it in the future.

Regarding the logging info, this will be displayed always when there is a df visualization, but if you consider that it is OK to have a warning we can add it